crates/worker/src/oneshot/spawner.rs

```diff
@@ -64,4 +65,17 @@ where
 
         OneshotBridge::new(inner, rx)
     }
+
+    /// Spawns a Worker according to the given options. Returns `None` if failed.
+    pub fn spawn_with_options(mut self, options: SpawnOptions<'_>) -> Option<OneshotBridge<N>>
```

Comment:
To simplify the API, most gloo methods have adopted panicking instead of `Error`.
If a spawning has failed, it should panic like other spawning methods.

There is no guarantee that spawning is going to success even if the API call is successful as spawning process is asynchronous.

Reply:
I think if that is the direction we decide to go, then we should update the API properly.
I would prefer dedicated PRs that revisits the API on a per-crate basis and not trying to sneak in with other PRs.

Reply:
I agree. Panicking is fine for this PR